src/test/emulators/functionsEmulator.spec.ts

```diff
@@ -54,7 +54,51 @@ describe("FunctionsEmulator-Hub", () => {
       });
   }).timeout(TIMEOUT_LONG);
 
-  it("should rewrite req.path to hide /:project_id/:trigger_id", async () => {
+  it("should route requests to /:project_id/:region/:trigger_id/ to HTTPS Function", async () => {
+    UseFunctions(() => {
+      require("firebase-admin").initializeApp();
+      return {
+        function_id: require("firebase-functions").https.onRequest(
+          (req: express.Request, res: express.Response) => {
+            res.json({ hello: "world" });
+          }
+        ),
+      };
+    });
+
+    await supertest(
+      FunctionsEmulator.createHubServer(FunctionRuntimeBundles.template, process.execPath)
+    )
+      .get("/fake-project-id/us-central-1f/function_id/")
+      .expect(200)
+      .then((res) => {
+        expect(res.body).to.deep.equal({ hello: "world" });
+      });
+  }).timeout(TIMEOUT_LONG);
+
+  it("should route requests to /:project_id/:region/:trigger_id/a/b to HTTPS Function", async () => {
+    UseFunctions(() => {
+      require("firebase-admin").initializeApp();
+      return {
+        function_id: require("firebase-functions").https.onRequest(
+          (req: express.Request, res: express.Response) => {
+            res.json({ hello: "world" });
+          }
+        ),
+      };
+    });
+
+    await supertest(
```

Comment:
Not your fault but while you're in here could we have tests to make sure that:

  * The HTTP method is preserved
  * Query params are preserved

Reply:
We already have a test for query params, do you think we need to check that in every test?

Reply:
Whoops sorry didn't know we already had that test.

Reply:
Was it failing back when we were dropping query params?  Because that's a pretty recent fix.

Reply:
I believe the test was added after that fix so we should be good.